07_cmd_lab/cmdlab.txt

```diff
@@ -1,13 +1,44 @@
-Intro To Command Line Lab.
-Questions online: https://www.tocode.co.il/bundles/unix1/lessons/cmdlab
-
-1. Login to your UNIX account. How do you find:
-  - your current shell?
-  - your default shell?
-  - a list of all shells currently installed on the machine?
-
-2. View the file /etc/passwd on your machine with the command "cat /etc/passwd". Then read about the file in "man 5 passwd". What information is held in the file? What are the fields in each line?
-
-3. In Unix, the /etc folder holds configuration files. Display all files in "/etc" folder ordered by file size (Hint: search in "man ls")
-
-4. Print current date/time formatted as: "Sun 21/03/2010, 14:30". Hint: read "man date" and search for the correct format string.
+Intro To Command Line Lab.
+Questions online: https://www.tocode.co.il/bundles/unix1/lessons/cmdlab
+
+Ophir Karako
+
+1. Login to your UNIX account. How do you find:
+  - your current shell?
+  echo $0
+  
+  - your default shell?
+  echo $SHELL
+  
+  - a list of all shells currently installed on the machine?
+  cat /etc/shells
+  
+  
+
+2. View the file /etc/passwd on your machine with the command "cat /etc/passwd". Then read about the file in "man 5 passwd". 
+What information is held in the file? What are the fields in each line?
+
+contains one line for each user account, with seven fields delimited by colons (“:”). These
+ +       fields are:
+ +       ·   login name
+ +       ·   optional encrypted password
+ +       ·   numerical user ID
+ +       ·   numerical group ID
+ +       ·   user name or comment field
+ +       ·   user home directory
+ +       ·   optional user command interpreter
+
+
+
+
+3. In Unix, the /etc folder holds configuration files. 
+Display all files in "/etc" folder ordered by file size (Hint: search in "man ls")
+
+ls -sort -S /etc

```

Comment:
ha! :smiley:  there is a subtle "bug" here (and I'm kinda glad you made it!) here let me attempt to explain:

what happens here is this:

``ls -s -o -r -r -S``

why? because there is a nice feature with command flags, it's possible to combine them, for example:

```ls -lh``` is the same as ```ls -l -h```

world flags are prefixed with "--", e,g: ```ls --all``` 

it's kinda awesome that ls has all  flags in the string "sort" :smile: